src/transcode/utils.ts

```diff
@@ -0,0 +1,194 @@
+import { BigNumberish } from 'ethers';
+import { ParamType } from 'ethers/lib/utils';
+import { CLIError } from '../utils/errors';
+import { readFileSync } from 'fs';
+import prompts from 'prompts';
+import { AST } from '../ast/ast';
+import { FunctionDefinition, SourceUnit } from 'solc-typed-ast';
+import { ABIEncoderVersion } from 'solc-typed-ast/dist/types/abi';
+import { divmod } from '../utils/utils';
+
+export type SolValue = BigNumberish | boolean | string | { [key: string]: SolValue } | SolValue[];
+export type Param = (string | number | Param)[];
+
+export function getWidthInFeltsOf(type: ParamType): number {
+  if (type.baseType.startsWith('uint')) {
+    const width = parseInt(type.baseType.slice(4), 10);
+    return width < 256 ? 1 : 2;
+  } else if (type.baseType.startsWith('int')) {
+    const width = parseInt(type.baseType.slice(3), 10);
+    return width < 256 ? 1 : 2;
+  } else if (type.baseType.startsWith('address')) {
+    return 1;
+  } else if (type.baseType.startsWith('bool')) {
+    return 1;
+  } else if (/byte\d*$/.test(type.baseType)) {
+    const width = parseInt(type.baseType.slice(4), 10);
+    return width * 8;
```

Comment:
Is there an error in the regex here? Shoudn't it be:
```typescript
/bytes\d*$/
```
Also the felt size for fixed bytes is 2 if fixed size is 32 and 1 in other case

Reply:
I just realized method `getWidthInFeltsOf` containing this is not being called. I am not sure what is it for. However, I will fix this and ask Swapnil